[BUG] old pandas not compatible with Python 3.13

### Reproduction steps

Running Fedora 41 which has Python 3.13. Pandas has known compatibility issues with Python 3.13 which has been fixed in pandas latest 2.2.3. Request bump in dependency versions.

Error includes failure to build "pandas/_libs/tslibs/ccalendar.cpython-313-x86_64-linux-gnu.so.p/meson-generated_pandas__libs_tslibs_ccalendar.pyx.c.o"

workaround to bump version
```
diff --git a/scripts/setup/constraints.txt b/scripts/setup/constraints.txt
index 874d46f6..4534c71a 100644
--- a/scripts/setup/constraints.txt
+++ b/scripts/setup/constraints.txt
@@ -141,7 +141,7 @@ packaging==23.0
     #   ghapi
     #   idf-component-manager
     #   west
-pandas==2.1.4 ; platform_machine != "aarch64" and platform_machine != "arm64"
+pandas==2.2.3 ; platform_machine != "aarch64" and platform_machine != "arm64"
     # via -r requirements.memory.txt
 parso==0.8.3
     # via jedi
```

install output with error
```
➜  esp-matter2 git:(main) cd ./connectedhomeip/connectedhomeip
./scripts/checkout_submodules.py --platform esp32 linux --shallow
Checking out: nlassert, nlio, mbedtls, qrcode, m5stack-tft, pigweed, openthread, ot-br-posix, cirque, nanopb, freertos, third_party/jsoncpp/repo, editline, third_party/boringssl/repo/src, third_party/libwebsockets/repo, third_party/imgui/repo, perfetto, third_party/lwip/repo, third_p
arty/abseil-cpp/src, third_party/fuzztest, third_party/googletest, third_party/re2/src
➜  connectedhomeip git:(9b8fffe0) cd ../..
./install.sh
Running Matter Setup

2025-02-21 13:35:50,791 Loading extra packages for linux
2025-02-21 13:35:50,791 Skipping: darwin (i.e. /home/achasen/esp/esp-matter2/connectedhomeip/connectedhomeip/third_party/pigweed/repo/pw_env_setup/py/pw_env_setup/cipd_setup/python311.json)
2025-02-21 13:35:50,791 Skipping: windows (i.e. /home/achasen/esp/esp-matter2/connectedhomeip/connectedhomeip/third_party/pigweed/repo/pw_env_setup/py/pw_env_setup/cipd_setup/python311.json)

  WELCOME TO...

         █
         █
     ▄   █   ▄                                █     █
     ▀▀█████▀▀      ▄▀▀▀▄ ▄▀▀▀▄    ▄▀▀▀▀▄█  ▀▀█▀▀▀▀▀█▀▀   ▄▀▀▀▀▄    ▄▀▀
   ▀█▄       ▄█▀   █     █     █  █      █    █     █    █▄▄▄▄▄▄█  █
     ▀█▄   ▄█▀     █     █     █  █      █    █     █    █         █
  ▄██▀▀█   █▀▀██▄  █     █     █   ▀▄▄▄▄▀█    ▀▄▄   ▀▄▄   ▀▄▄▄▄▀   █
 ▀▀    █   █    ▀▀

  BOOTSTRAP! Bootstrap may take a few minutes; please be patient.

Downloading and installing packages into local source directory:

  Setting up CIPD package manager...done (42.7s)
  Setting up Project actions........skipped (0.1s)
  Setting up Python environment.....done (2m0.2s)
  Setting up pw packages............skipped (0.1s)
  Setting up Host tools.............done (0.1s)

Activating environment (setting environment variables):

  Setting environment variables for CIPD package manager...done
  Setting environment variables for Project actions........skipped
  Setting environment variables for Python environment.....done
  Setting environment variables for pw packages............skipped
  Setting environment variables for Host tools.............done

Checking the environment:

20250221 13:38:34 INF Environment passes all checks!

Environment looks good, you are ready to go!

To reactivate this environment in the future, run this in your
terminal:

  source ./activate.sh

To deactivate this environment, run this:

  deactivate

Installing pip requirements for all...
  error: subprocess-exited-with-error

  × Preparing metadata (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [124 lines of output]
      + meson setup /tmp/pip-install-zdyk2giy/pandas_c138ace9c29d4e859528ec18aa31104e /tmp/pip-install-zdyk2giy/pandas_c138ace9c29d4e859528ec18aa31104e/.mesonpy-51b25gii/build -Dbuildtype=release -Db_ndebug=if-release -Db_vscrt=md --vsenv --native-file=/tmp/pip-install-zdyk2giy/panda
s_c138ace9c29d4e859528ec18aa31104e/.mesonpy-51b25gii/build/meson-python-native-file.ini
      The Meson build system
      Version: 1.2.1
      Source dir: /tmp/pip-install-zdyk2giy/pandas_c138ace9c29d4e859528ec18aa31104e
      Build dir: /tmp/pip-install-zdyk2giy/pandas_c138ace9c29d4e859528ec18aa31104e/.mesonpy-51b25gii/build
      Build type: native build
      Project name: pandas
      Project version: 2.1.4
      C compiler for the host machine: ccache cc (gcc 14.2.1 "cc (GCC) 14.2.1 20250110 (Red Hat 14.2.1-7)")
      C linker for the host machine: cc ld.bfd 2.43.1-5
      C++ compiler for the host machine: ccache c++ (gcc 14.2.1 "c++ (GCC) 14.2.1 20250110 (Red Hat 14.2.1-7)")
      C++ linker for the host machine: c++ ld.bfd 2.43.1-5
      Cython compiler for the host machine: cython (cython 0.29.37)
      Host machine cpu family: x86_64
      Host machine cpu: x86_64
      Program python found: YES (/home/achasen/esp/esp-matter2/connectedhomeip/connectedhomeip/.environment/pigweed-venv/bin/python)
      Found pkg-config: /usr/bin/pkg-config (2.3.0)
      Run-time dependency python found: YES 3.13
      Build targets in project: 53

      pandas 2.1.4

        User defined options
          Native files: /tmp/pip-install-zdyk2giy/pandas_c138ace9c29d4e859528ec18aa31104e/.mesonpy-51b25gii/build/meson-python-native-file.ini
          buildtype   : release
          vsenv       : True
          b_ndebug    : if-release
          b_vscrt     : md

      Found ninja-1.13.0.git.fuchsia at /home/achasen/esp/esp-matter2/connectedhomeip/connectedhomeip/.environment/cipd/packages/pigweed/bin/ninja

      Visual Studio environment is needed to run Ninja. It is recommended to use Meson wrapper:
      /tmp/pip-build-env-05xua6wh/overlay/bin/meson compile -C .
      + /home/achasen/esp/esp-matter2/connectedhomeip/connectedhomeip/.environment/cipd/packages/pigweed/bin/ninja
      [1/151] Generating pandas/_libs/algos_common_helper_pxi with a custom command
      [2/151] Generating pandas/_libs/algos_take_helper_pxi with a custom command
      [3/151] Generating pandas/_libs/hashtable_func_helper_pxi with a custom command
      [4/151] Generating pandas/_libs/khash_primitive_helper_pxi with a custom command
      [5/151] Generating pandas/_libs/hashtable_class_helper_pxi with a custom command
      [6/151] Generating pandas/_libs/index_class_helper_pxi with a custom command
      [7/151] Generating pandas/_libs/intervaltree_helper_pxi with a custom command
      [8/151] Generating pandas/_libs/sparse_op_helper_pxi with a custom command
      [9/151] Compiling Cython source /tmp/pip-install-zdyk2giy/pandas_c138ace9c29d4e859528ec18aa31104e/pandas/_libs/tslibs/ccalendar.pyx
      [10/151] Compiling Cython source /tmp/pip-install-zdyk2giy/pandas_c138ace9c29d4e859528ec18aa31104e/pandas/_libs/tslibs/nattype.pyx
      [11/151] Compiling Cython source /tmp/pip-install-zdyk2giy/pandas_c138ace9c29d4e859528ec18aa31104e/pandas/_libs/tslibs/base.pyx
      [12/151] Compiling Cython source /tmp/pip-install-zdyk2giy/pandas_c138ace9c29d4e859528ec18aa31104e/pandas/_libs/arrays.pyx
      [13/151] Compiling Cython source /tmp/pip-install-zdyk2giy/pandas_c138ace9c29d4e859528ec18aa31104e/pandas/_libs/tslibs/dtypes.pyx
      [14/151] Compiling Cython source /tmp/pip-install-zdyk2giy/pandas_c138ace9c29d4e859528ec18aa31104e/pandas/_libs/indexing.pyx
      [15/151] Compiling Cython source /tmp/pip-install-zdyk2giy/pandas_c138ace9c29d4e859528ec18aa31104e/pandas/_libs/tslibs/tzconversion.pyx
      [16/151] Compiling Cython source /tmp/pip-install-zdyk2giy/pandas_c138ace9c29d4e859528ec18aa31104e/pandas/_libs/tslibs/np_datetime.pyx
      [17/151] Compiling Cython source /tmp/pip-install-zdyk2giy/pandas_c138ace9c29d4e859528ec18aa31104e/pandas/_libs/tslibs/timezones.pyx
      [18/151] Compiling Cython source /tmp/pip-install-zdyk2giy/pandas_c138ace9c29d4e859528ec18aa31104e/pandas/_libs/tslibs/strptime.pyx
      [19/151] Compiling Cython source /tmp/pip-install-zdyk2giy/pandas_c138ace9c29d4e859528ec18aa31104e/pandas/_libs/tslibs/conversion.pyx
      [20/151] Compiling Cython source /tmp/pip-install-zdyk2giy/pandas_c138ace9c29d4e859528ec18aa31104e/pandas/_libs/tslibs/vectorized.pyx
      [21/151] Compiling Cython source /tmp/pip-install-zdyk2giy/pandas_c138ace9c29d4e859528ec18aa31104e/pandas/_libs/tslibs/offsets.pyx
      [22/151] Compiling Cython source /tmp/pip-install-zdyk2giy/pandas_c138ace9c29d4e859528ec18aa31104e/pandas/_libs/tslibs/timedeltas.pyx
      [23/151] Compiling Cython source /tmp/pip-install-zdyk2giy/pandas_c138ace9c29d4e859528ec18aa31104e/pandas/_libs/tslibs/parsing.pyx
      [24/151] Compiling Cython source /tmp/pip-install-zdyk2giy/pandas_c138ace9c29d4e859528ec18aa31104e/pandas/_libs/ops_dispatch.pyx
      [25/151] Compiling Cython source /tmp/pip-install-zdyk2giy/pandas_c138ace9c29d4e859528ec18aa31104e/pandas/_libs/tslibs/fields.pyx
      [26/151] Compiling Cython source /tmp/pip-install-zdyk2giy/pandas_c138ace9c29d4e859528ec18aa31104e/pandas/_libs/hashing.pyx
      [27/151] Compiling Cython source /tmp/pip-install-zdyk2giy/pandas_c138ace9c29d4e859528ec18aa31104e/pandas/_libs/tslibs/period.pyx
      [28/151] Compiling Cython source /tmp/pip-install-zdyk2giy/pandas_c138ace9c29d4e859528ec18aa31104e/pandas/_libs/tslibs/timestamps.pyx
      [29/151] Compiling Cython source /tmp/pip-install-zdyk2giy/pandas_c138ace9c29d4e859528ec18aa31104e/pandas/_libs/properties.pyx
      [30/151] Compiling Cython source /tmp/pip-install-zdyk2giy/pandas_c138ace9c29d4e859528ec18aa31104e/pandas/_libs/missing.pyx
      [31/151] Compiling Cython source /tmp/pip-install-zdyk2giy/pandas_c138ace9c29d4e859528ec18aa31104e/pandas/_libs/byteswap.pyx
      [32/151] Compiling Cython source /tmp/pip-install-zdyk2giy/pandas_c138ace9c29d4e859528ec18aa31104e/pandas/_libs/internals.pyx
      [33/151] Compiling Cython source /tmp/pip-install-zdyk2giy/pandas_c138ace9c29d4e859528ec18aa31104e/pandas/_libs/testing.pyx
      [34/151] Compiling C object pandas/_libs/tslibs/base.cpython-313-x86_64-linux-gnu.so.p/meson-generated_pandas__libs_tslibs_base.pyx.c.o
      FAILED: pandas/_libs/tslibs/base.cpython-313-x86_64-linux-gnu.so.p/meson-generated_pandas__libs_tslibs_base.pyx.c.o
      ccache cc -Ipandas/_libs/tslibs/base.cpython-313-x86_64-linux-gnu.so.p -Ipandas/_libs/tslibs -I../../pandas/_libs/tslibs -I../../../../pip-build-env-05xua6wh/overlay/lib64/python3.13/site-packages/numpy/core/include -I../../pandas/_libs/include -I/usr/include/python3.13 -fvisib
ility=hidden -fdiagnostics-color=always -DNDEBUG -D_FILE_OFFSET_BITS=64 -w -std=c99 -O3 -DNPY_NO_DEPRECATED_API=0 -DNPY_TARGET_VERSION=NPY_1_21_API_VERSION -fPIC -MD -MQ pandas/_libs/tslibs/base.cpython-313-x86_64-linux-gnu.so.p/meson-generated_pandas__libs_tslibs_base.pyx.c.o -MF pa
ndas/_libs/tslibs/base.cpython-313-x86_64-linux-gnu.so.p/meson-generated_pandas__libs_tslibs_base.pyx.c.o.d -o pandas/_libs/tslibs/base.cpython-313-x86_64-linux-gnu.so.p/meson-generated_pandas__libs_tslibs_base.pyx.c.o -c pandas/_libs/tslibs/base.cpython-313-x86_64-linux-gnu.so.p/pan
das/_libs/tslibs/base.pyx.c
      pandas/_libs/tslibs/base.cpython-313-x86_64-linux-gnu.so.p/pandas/_libs/tslibs/base.pyx.c: In function ‘__Pyx_PyInt_As_long’:
      pandas/_libs/tslibs/base.cpython-313-x86_64-linux-gnu.so.p/pandas/_libs/tslibs/base.pyx.c:5397:27: error: too few arguments to function ‘_PyLong_AsByteArray’
       5397 |                 int ret = _PyLong_AsByteArray((PyLongObject *)v,   

      pandas/_libs/tslibs/base.cpython-313-x86_64-linux-gnu.so.p/pandas/_libs/tslibs/base.pyx.c: In function ‘__Pyx_PyInt_As_long’:
      pandas/_libs/tslibs/base.cpython-313-x86_64-linux-gnu.so.p/pandas/_libs/tslibs/base.pyx.c:5397:27: error: too few arguments to function ‘_PyLong_AsByteArray’
       5397 |                 int ret = _PyLong_AsByteArray((PyLongObject *)v,
            |                           ^~~~~~~~~~~~~~~~~~~
      In file included from /usr/include/python3.13/longobject.h:107,
                       from /usr/include/python3.13/Python.h:81,
                       from pandas/_libs/tslibs/base.cpython-313-x86_64-linux-gnu.so.p/pandas/_libs/tslibs/base.pyx.c:6:
      /usr/include/python3.13/cpython/longobject.h:111:17: note: declared here
        111 | PyAPI_FUNC(int) _PyLong_AsByteArray(PyLongObject* v,
            |                 ^~~~~~~~~~~~~~~~~~~
      pandas/_libs/tslibs/base.cpython-313-x86_64-linux-gnu.so.p/pandas/_libs/tslibs/base.pyx.c: In function ‘__Pyx_PyInt_As_int’:
      pandas/_libs/tslibs/base.cpython-313-x86_64-linux-gnu.so.p/pandas/_libs/tslibs/base.pyx.c:5631:27: error: too few arguments to function ‘_PyLong_AsByteArray’
       5631 |                 int ret = _PyLong_AsByteArray((PyLongObject *)v,
            |                           ^~~~~~~~~~~~~~~~~~~
      /usr/include/python3.13/cpython/longobject.h:111:17: note: declared here
        111 | PyAPI_FUNC(int) _PyLong_AsByteArray(PyLongObject* v,
            |                 ^~~~~~~~~~~~~~~~~~~
      [35/151] Compiling C object pandas/_libs/tslibs/ccalendar.cpython-313-x86_64-linux-gnu.so.p/meson-generated_pandas__libs_tslibs_ccalendar.pyx.c.o
      FAILED: pandas/_libs/tslibs/ccalendar.cpython-313-x86_64-linux-gnu.so.p/meson-generated_pandas__libs_tslibs_ccalendar.pyx.c.o
      ccache cc -Ipandas/_libs/tslibs/ccalendar.cpython-313-x86_64-linux-gnu.so.p -Ipandas/_libs/tslibs -I../../pandas/_libs/tslibs -I../../../../pip-build-env-05xua6wh/overlay/lib64/python3.13/site-packages/numpy/core/include -I../../pandas/_libs/include -I/usr/include/python3.13 -fvisibility=hidden -fdiagnostics-color=always -DNDEBUG -D_FILE_OFFSET_BITS=64 -w -std=c99 -O3 -DNPY_NO_DEPRECATED_API=0 -DNPY_TARGET_VERSION=NPY_1_21_API_VERSION -fPIC -MD -MQ pandas/_libs/tslibs/ccalendar.cpython-313-x86_64-linux-gnu.so.p/meson-generated_pandas__libs_tslibs_ccalendar.pyx.c.o -MF pandas/_libs/tslibs/ccalendar.cpython-313-x86_64-linux-gnu.so.p/meson-generated_pandas__libs_tslibs_ccalendar.pyx.c.o.d -o pandas/_libs/tslibs/ccalendar.cpython-313-x86_64-linux-gnu.so.p/meson-generated_pandas__libs_tslibs_ccalendar.pyx.c.o -c pandas/_libs/tslibs/ccalendar.cpython-313-x86_64-linux-gnu.so.p/pandas/_libs/tslibs/ccalendar.pyx.c
      pandas/_libs/tslibs/ccalendar.cpython-313-x86_64-linux-gnu.so.p/pandas/_libs/tslibs/ccalendar.pyx.c: In function ‘__Pyx_PyInt_As_int’:
      pandas/_libs/tslibs/ccalendar.cpython-313-x86_64-linux-gnu.so.p/pandas/_libs/tslibs/ccalendar.pyx.c:7376:27: error: too few arguments to function ‘_PyLong_AsByteArray’
       7376 |                 int ret = _PyLong_AsByteArray((PyLongObject *)v,
            |                           ^~~~~~~~~~~~~~~~~~~
      In file included from /usr/include/python3.13/longobject.h:107,
                       from /usr/include/python3.13/Python.h:81,
                       from pandas/_libs/tslibs/ccalendar.cpython-313-x86_64-linux-gnu.so.p/pandas/_libs/tslibs/ccalendar.pyx.c:6:
      /usr/include/python3.13/cpython/longobject.h:111:17: note: declared here
        111 | PyAPI_FUNC(int) _PyLong_AsByteArray(PyLongObject* v,
            |                 ^~~~~~~~~~~~~~~~~~~
      pandas/_libs/tslibs/ccalendar.cpython-313-x86_64-linux-gnu.so.p/pandas/_libs/tslibs/ccalendar.pyx.c: In function ‘__Pyx_PyInt_As_long’:
      pandas/_libs/tslibs/ccalendar.cpython-313-x86_64-linux-gnu.so.p/pandas/_libs/tslibs/ccalendar.pyx.c:7686:27: error: too few arguments to function ‘_PyLong_AsByteArray’
       7686 |                 int ret = _PyLong_AsByteArray((PyLongObject *)v,
            |                           ^~~~~~~~~~~~~~~~~~~
      /usr/include/python3.13/cpython/longobject.h:111:17: note: declared here
        111 | PyAPI_FUNC(int) _PyLong_AsByteArray(PyLongObject* v,
            |                 ^~~~~~~~~~~~~~~~~~~
      [36/151] Compiling Cython source /tmp/pip-install-zdyk2giy/pandas_c138ace9c29d4e859528ec18aa31104e/pandas/_libs/ops.pyx
      [37/151] Compiling Cython source /tmp/pip-install-zdyk2giy/pandas_c138ace9c29d4e859528ec18aa31104e/pandas/_libs/tslib.pyx
      [38/151] Compiling Cython source /tmp/pip-install-zdyk2giy/pandas_c138ace9c29d4e859528ec18aa31104e/pandas/_libs/index.pyx
      [39/151] Compiling Cython source /tmp/pip-install-zdyk2giy/pandas_c138ace9c29d4e859528ec18aa31104e/pandas/_libs/window/indexers.pyx
      [40/151] Compiling Cython source /tmp/pip-install-zdyk2giy/pandas_c138ace9c29d4e859528ec18aa31104e/pandas/_libs/sas.pyx
      [41/151] Compiling Cython source /tmp/pip-install-zdyk2giy/pandas_c138ace9c29d4e859528ec18aa31104e/pandas/_libs/reshape.pyx
      [42/151] Compiling Cython source /tmp/pip-install-zdyk2giy/pandas_c138ace9c29d4e859528ec18aa31104e/pandas/_libs/writers.pyx
      [43/151] Compiling Cython source /tmp/pip-install-zdyk2giy/pandas_c138ace9c29d4e859528ec18aa31104e/pandas/_libs/parsers.pyx
      [44/151] Compiling Cython source /tmp/pip-install-zdyk2giy/pandas_c138ace9c29d4e859528ec18aa31104e/pandas/_libs/lib.pyx
      [45/151] Compiling Cython source /tmp/pip-install-zdyk2giy/pandas_c138ace9c29d4e859528ec18aa31104e/pandas/_libs/interval.pyx
      [46/151] Compiling Cython source /tmp/pip-install-zdyk2giy/pandas_c138ace9c29d4e859528ec18aa31104e/pandas/_libs/window/aggregations.pyx
      [47/151] Compiling Cython source /tmp/pip-install-zdyk2giy/pandas_c138ace9c29d4e859528ec18aa31104e/pandas/_libs/sparse.pyx
      [48/151] Compiling Cython source /tmp/pip-install-zdyk2giy/pandas_c138ace9c29d4e859528ec18aa31104e/pandas/_libs/hashtable.pyx
      [49/151] Compiling Cython source /tmp/pip-install-zdyk2giy/pandas_c138ace9c29d4e859528ec18aa31104e/pandas/_libs/algos.pyx
      [50/151] Compiling Cython source /tmp/pip-install-zdyk2giy/pandas_c138ace9c29d4e859528ec18aa31104e/pandas/_libs/join.pyx
      [51/151] Compiling Cython source /tmp/pip-install-zdyk2giy/pandas_c138ace9c29d4e859528ec18aa31104e/pandas/_libs/groupby.pyx
      ninja: build stopped: subcommand failed.
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.

[notice] A new release of pip is available: 23.2.1 -> 25.0.1
[notice] To update, run: pip install --upgrade pip
```

### Bug prevalence

always

### GitHub hash of the SDK that was being used

9b8fffe0bb4e7ba7aac319f5905070a3476db7cb

### Platform

esp32

### Platform Version(s)

_No response_

### Anything else?

_No response_

--- a/scripts/setup/constraints.txt
+++ b/scripts/setup/constraints.txt
@@ -141,7 +141,7 @@ packaging==23.0
     #   ghapi
     #   idf-component-manager
     #   west
-pandas==2.1.4 ; platform_machine != "aarch64" and platform_machine != "arm64"
+pandas==2.2.3 ; platform_machine != "aarch64" and platform_machine != "arm64"
     # via -r requirements.memory.txt
 parso==0.8.3
     # via jedi

